core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -73,8 +73,14 @@
             // 2. initiate RudderElementCache
             RudderLogger.logDebug("EventRepository: constructor: Initiating RudderElementCache");
             RudderElementCache.initiate(_application);
+            preferenceManager = RudderPreferenceManager.getInstance(_application);
+            String anonymousId;
+            RudderLogger.logDebug(RudderElementCache.getCachedContext().getAnonymousId());
+            if (RudderElementCache.getCachedContext().getAnonymousId() == null)
```

Comment:
```suggestion
            if (RudderElementCache.getCachedContext().getAnonymousId() == null) {
            	// block
            }
```
for better redability

Reply:
done